src/cnaas_nms/api/generic.py

```diff
@@ -56,6 +58,51 @@ def offset_results() -> int:
     return offset
 
 
+def pagination_headers(total_count) -> dict:
+    per_page = DEFAULT_PER_PAGE
+    page_arg = 1
+    links = []
+    headers = {
+        "X-Total-Count": total_count,
+    }
+
+    args = request.args
+    if 'per_page' in args:
+        try:
+            per_page_arg = int(args['per_page'])
+            assert 1 <= per_page_arg <= MAX_PER_PAGE
+            per_page = per_page_arg
+        except (AssertionError, ValueError):
+            pass
+
+    last_page = int(math.ceil(total_count / per_page))
+    if last_page == 1:
+        return {}
+
+    if 'page' in args:
+        try:
+            page_arg = max(1, int(args['page']))
+        except ValueError:
+            pass
+
+    query_without_page = {k: request.args[k] for k in request.args if k not in ["page"]}
+
+    if page_arg * per_page < total_count:
```

Comment:
This conditional could use last_page: `if page_arg < last_page`

Reply:
Yes that makes more sense!